### PR TITLE
fixup a few ruby warnings

### PR DESCRIPTION
**Pull request by eprothro**
We author a gem that has the ruby-driver as a dependency.

These are fixups for a few of the warnings that are are most repeatedly reported when we run our suite.

Happy to hear y'alls thoughts if we're being pedantic here, but we try to run our unit suite with warnings enabled to help fight entropy.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -5,18 +5,30 @@
 All code has bugs, but if you report them they can be squashed.
 
 The best bug reports include everything that is needed to reliably reproduce the bug.
+
+### Running the Test Suite
+
 Try to write a test case and include it in your report (have a look at the
-[regression test suite](spec/integration/regression_spec.rb) if you need inspiration).
-Submit defect reports to our [Jira](https://datastax-oss.atlassian.net/projects/RUBY/issues).
+[regression test suite](spec/regressions/**/*_spec.rb) if you need inspiration).
```
Comment (avalanche123) [inline CONTRIBUTING.md]: this breaks github links.

File: lib/cassandra/cluster/control_connection.rb [hunk 2]
```diff
@@ -825,7 +826,7 @@ def refresh_schema_async_wrapper
         # 2. avoid race conditions where those updates may return older data than our
         #    full refresh and might win as last writer with that potentially older data.
         synchronize do
-          if @refresh_schema_future
+          if defined?(@refresh_schema_future) && @refresh_schema_future
```
Comment (avalanche123) [inline lib/cassandra/cluster/control_connection.rb]: maybe initialize to `nil` instead of using `defined?`

File: lib/cassandra/executors.rb [hunk 3]
```diff
@@ -80,6 +80,7 @@ def run
 
         loop do
           tasks = nil
+          @term ||= nil
```
Comment (avalanche123) [inline lib/cassandra/executors.rb]: this is already initialize in constructor

File: CONTRIBUTING.md (hunk 1 above)
Comment (eprothro) [inline CONTRIBUTING.md]: @avalanche123 thanks

File: lib/cassandra/cluster/control_connection.rb (hunk 2 above)
Comment (eprothro) [inline lib/cassandra/cluster/control_connection.rb]: @avalanche123 happy to. `@refresh_schema_future` seemed much more private than the rest of the variables in the initializer. You prefer that?

File: lib/cassandra/executors.rb (hunk 3 above)
Comment (eprothro) [inline lib/cassandra/executors.rb]: @avalanche123 I saw that, but I still get the warning as a consumer. "Something something threading something"?

File: lib/cassandra/cluster/control_connection.rb (hunk 2 above)
Comment (avalanche123) [inline lib/cassandra/cluster/control_connection.rb]: yeah, explicit initialization seems better

Comment (eprothro) [inline lib/cassandra/cluster/control_connection.rb]: @avalanche123 done, thanks.

File: lib/cassandra/executors.rb (hunk 3 above)
Comment (avalanche123) [inline lib/cassandra/executors.rb]: I think you should be able to get away with moving `@term = ...` above `@pool = ...` in the constructor and remove `@term ||= nil` let me know if it works

**Comment by avalanche123** [comment]
+1 after some corrections

File: lib/cassandra/executors.rb (hunk 3 above)
Comment (eprothro) [inline lib/cassandra/executors.rb]: @avalanche123 ah, makes sense, thanks -- I should have looked more closely.

**Comment by eprothro** [comment]
corrections pushed and squashed.

**Comment by avalanche123** [comment]
thanks!

**Comment by eprothro** [comment]
thank *you*!

**Pull request merged**